storage/innobase/buf/buf0buf.cc

```diff
@@ -1878,6 +2095,96 @@ inline void buf_pool_t::resize()
 	return;
 }
 
+static std::mutex garbage_collect_mutex{};
+
+inline void buf_pool_t::garbage_collect()
```

Comment:
This code could be inside `#ifdef __linux__` because it is only being used on Linux.